.github/workflows/clang_format.yml

```diff
@@ -0,0 +1,17 @@
+name: clang-format
+
+on:
+  merge_group:
+  pull_request:
+  push:
+    branches: [main]
+
+jobs:
+  formatting-check:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+    - name: Check formatting
+      uses: jidicula/clang-format-action@v4.13.0
+      with:
+        clang-format-version: '14' # TODO?
```

Comment:
What version should this have? Empirically 14 was the first version that matched your current formatting and did not complain about unknown configuration options, but this can go all the way up to 18 if needed.

Reply:
What is a reasonable minimum version? I am not a big Clang user personally - I have clang-18 locally.
Split the difference and use 16?